src/spec-configuration/containerFeaturesConfiguration.ts

```diff
@@ -256,12 +256,21 @@ export function getFeatureInstallWrapperScript(feature: Feature, featureSet: Fea
 	const documentation = escapeQuotesForShell(feature.documentationURL ?? '');
 	const optionsIndented = escapeQuotesForShell(options.map(x => `    ${x}`).join('\n'));
 
+	let warningHeader = '';
+	if (feature.deprecated) {
+		warningHeader += `(!) WARNING: Using the deprecated Feature "${escapeQuotesForShell(feature.id)}". This Feature will no longer receive any further updates/support.\n`;
+	}
+
+	if (feature?.legacyIds && feature.legacyIds.length > 0 && feature.currentId && feature.id !== feature.currentId) {
+		warningHeader += `(!) WARNING: This feature has been renamed. Please update the reference in devcontainer.json to "${escapeQuotesForShell(feature.currentId)}".`;
+	}
+
 	const errorMessage = `ERROR: Feature "${name}" (${id}) failed to install!`;
 	const troubleshootingMessage = documentation
 		? ` Look at the documentation at ${documentation} for help troubleshooting this error.`
 		: '';
 
-	return `#!/bin/sh
+	return `#!/bin/bash
```

Comment:
Can we keep this as `sh`? I had originally used `sh` here because it's available in more containers and unless we really need bash support, it would be good to use the most broadly-available shell. Some alpine containers don't have bash by default as an example and it would be good to support those as much as possible.

One way to do this would be to avoid the conditional in the script itself and put the conditional in the string concatenation that prints out the script.

Reply:
👍  Used an sh condition statement instead. https://github.com/devcontainers/cli/actions/runs/3849044939/jobs/6557564151#step:7:216

![image](https://user-images.githubusercontent.com/24955913/210845768-26c722a8-8abc-42b4-90ea-6f506ae8e670.png)